toolchain/check/eval.cpp

```diff
@@ -1311,6 +1324,41 @@ static auto MakeConstantForBuiltinCall(Context& context, SemIRLoc loc,
       return SemIR::ConstantId::NotConstant;
     }
 
+    case SemIR::BuiltinFunctionKind::TypeAnd: {
+      CARBON_CHECK(arg_ids.size() == 2);
+      auto facet_type_ids = std::to_array<SemIR::FacetTypeId>({
+          SemIR::FacetTypeId::None,
+          SemIR::FacetTypeId::None,
+      });
+      for (auto [i, arg_id] : llvm::enumerate(arg_ids)) {
+        if (auto facet_type =
+                context.insts().TryGetAs<SemIR::FacetType>(arg_id)) {
+          facet_type_ids[i] = facet_type->facet_type_id;
+        } else {
+          CARBON_DIAGNOSTIC(
+              FacetTypeRequiredForTypeAndOperator, Error,
+              "BitAnd operator on types requires facet type on both sides");
+          context.emitter().Emit(loc, FacetTypeRequiredForTypeAndOperator);
+        }
+      }
+      // Allow errors to be diagnosed for both sides of the operator before
+      // returning here if any error occurred on either side.
+      if (llvm::any_of(facet_type_ids,
+                       [](SemIR::FacetTypeId id) { return !id.has_value(); })) {
+        return SemIR::ConstantId::NotConstant;
+      }
```

Comment:
Should this be returning an ErrorInst on an issue?

Reply:
Oh, yes! TIL `SemIR::ErrorInst::SingletonConstantId` exists